### PR TITLE
Disable response compression for k8s restAPI in client-go

### PR DESCRIPTION
**Pull request by ckuduvalli**
Keda-operator timing out for large number of configmaps and secrets. Hence disabling compression in client-go.
https://github.com/kedacore/keda/issues/3863

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes https://github.com/kedacore/keda/issues/3863

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
Thanks for the fix.

1. Shouldn't we add the same to Metrics Server config? https://github.com/kedacore/keda/blob/dab87b52d5ad56aca439ad9f5683e7ad641a4c60/adapter/main.go#L75

2. Could you please add this issue to Changelog?
3. And also a document it, probably here: https://keda.sh/docs/2.9/operate/cluster/#kubernetes-client-parameters

**Comment by ckuduvalli** [comment]
have added documentation here:
https://github.com/kedacore/keda-docs/pull/978

**Comment by ckuduvalli** [comment]
@zroubalik all feedback incorporated. Please check

**Review by zroubalik — APPROVED** [review]
Looking good, thanks a lot!

File: adapter/main.go [hunk 1]
```diff
@@ -209,6 +211,7 @@ func main() {
 	cmd.Flags().StringVar(&prometheusMetricsPath, "metrics-path", "/metrics", "Set the path for the prometheus metrics endpoint")
 	cmd.Flags().Float32Var(&adapterClientRequestQPS, "kube-api-qps", 20.0, "Set the QPS rate for throttling requests sent to the apiserver")
 	cmd.Flags().IntVar(&adapterClientRequestBurst, "kube-api-burst", 30, "Set the burst for throttling requests sent to the apiserver")
+	cmd.Flags().BoolVar(&disableCompression, "disable-compression", false, "Disable response compression for k8s restAPI in client-go. ")
```
Comment (zroubalik) [inline adapter/main.go:214]: Actually, I think we should make this `true` by default for both Operator and Metrics Server

Comment (zroubalik) [reply adapter/main.go:214]: @kedacore/keda-core-contributors WDYT?

Comment (zroubalik) [reply adapter/main.go:214]: Let's try it out, we can always revert, @ckuduvalli could you please change the default value? We will run e2e test and if it is good, we will merge it (and update docs PR as well) Thanks 🙏

Comment (ckuduvalli) [reply adapter/main.go:214]: have changed the default value to true

Comment (ckuduvalli) [reply adapter/main.go]: have updated charts as well
https://github.com/kedacore/charts/pull/326

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**